test/common/grpc/grpc_web_filter_test.cc

```diff
@@ -0,0 +1,126 @@
+#include "common/buffer/buffer_impl.h"
+#include "common/common/base64.h"
+#include "common/grpc/grpc_web_filter.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/headers.h"
+
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/upstream/mocks.h"
+#include "test/test_common/printers.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::_;
+using testing::NiceMock;
+using testing::Invoke;
+
+namespace Envoy {
+namespace Grpc {
+namespace {
+const char MESSAGE[] = "\x00\x00\x00\x00\x11grpc-web-bin-data";
+const size_t MESSAGE_SIZE = 22;
+const char TEXT_MESSAGE[] = "\x00\x00\x00\x00\x12grpc-web-text-data";
+const size_t TEXT_MESSAGE_SIZE = 23;
+const char B64_MESSAGE[] = "AAAAABJncnBjLXdlYi10ZXh0LWRhdGE=";
+const size_t B64_MESSAGE_SIZE = 32;
+const char TRAILERS[] = "\x80\x00\x00\x00\x0fgrpc-status:0\r\n";
+const size_t TRAILERS_SIZE = 20;
```

Comment:
Can we use `sizeof` here to make this less fragile, since these are const arrays with known size?